src/java.management/share/classes/javax/management/ObjectName.java

```diff
@@ -2016,12 +2016,21 @@ public boolean apply(ObjectName name) {
     }
 
     private final boolean matchDomains(ObjectName name) {
+        boolean useOptimized = (this.getClass() == ObjectName.class) &&
+                               (name.getClass() == ObjectName.class);
+
         if (isDomainPattern()) {
             // wildmatch domains
             // This ObjectName is the pattern
             // The other ObjectName is the string.
+            if (useOptimized)
```

Comment:
2026 and 2031 I was referring to with the nit about if and { }

Reply:
> 2026 and 2031 I was referring to with the nit about if and { }

Hi @kevinjwalls ! Thank you for review.
The braces are added

Reply:
Thanks!